### DISCUSSION TITLE
Unknown column type "timestamp" when changing column

### DISCUSSION THREAD

**Discussion by dakujem** [laravel/framework#39275] [Q&A]
- PHP Version: `8.0.12`
- `doctrine/dbal          3.1.3` (also present on `3.1.1`)
- `illuminate/database    v8.65.0` (also present on `v8.55.0`)

### Description:

Changing `timestamp`-type column does not work during a migration.

I'm adding a comment to an existing column of type `timestamp`:
```php
        Schema::table('operations', function (Blueprint $table) {
            $table->timestamp('at')->comment('a comment')->change();
        });
```
and the resulting error is this:
```
Doctrine\DBAL\Exception: Unknown column type "timestamp" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.
```

It doesn't matter what the changes are, simply calling `change` on the column results in the error.

### Steps To Reproduce:

```
php artisan make:migration whatever
```
then define the `up` method as follows:
```php
    public function up(): void
    {
        // create the table
        Schema::create('operations', function (Blueprint $table) {
            $table->timestamp('at');
        });
        // then try to modify the column
        Schema::table('operations', function (Blueprint $table) {
            $table->timestamp('at')->comment('a comment')->change();
        });
    }
```

> Two separate migrations might be needed, I have not tried it in exactly this same fashion.

**Comment by dakujem** [comment]
I'm not sure why this issue has been moved to discussions. Is this the new disguised place to report issues?

@driesvints It's a report of something that does not work as advertised, a misconfiguration, I suspect. Not a question nor a feature request.

**Comment by dakujem** [comment]
UPDATE: doesn't work with `char` column type either, same issue: `Unknown column type "char" requested.`

**Reply by hafezdivandari** [reply]
This is already fixed on #41320

**Comment by hafezdivandari** [comment]
To modify a `timestamp` type you have to also add the following configuration to your application's `config/database.php` configuration file:

```
use Illuminate\Database\DBAL\TimestampType;
 
'dbal' => [
    'types' => [
        'timestamp' => TimestampType::class,
    ],
],
```

See the docs for more info: https://laravel.com/docs/9.x/migrations#prerequisites
